File: leetcode/max-profit.ts
function maxProfit(prices: number[]): number {
  let maxProfit = 0;
  let startPrice = prices[0];

  for (let i = 1; i < prices.length; i++) {
    const currentPrice = prices[i];
    if (currentPrice > startPrice) {
      const profit = currentPrice - startPrice;
      maxProfit = Math.max(maxProfit, profit);
    } else {
      startPrice = currentPrice;
    }
  }

  return maxProfit;
}

File: leetcode/max-profit.go
package leetcode

func maxProfit(prices []int) int {
	maxProfit := 0
	startPrice := prices[0]
	for i := 1; i < len(prices); i++ {
		if prices[i] < startPrice {
			startPrice = prices[i]
		} else {
			if prices[i]-startPrice > maxProfit {
				maxProfit = prices[i] - startPrice
			}
		}
	}
	return maxProfit
}
